crowbar_framework/lib/crowbar/validator/network_validator.rb

```diff
@@ -4,7 +4,7 @@ class NetworkValidator
       def validate_network
         `#{Rails.root.join("..", "bin/network-json-validator").to_s} \
           --admin-ip \
-          #{Socket.ip_address_list.detect(&:ipv4_private?).ip_address} \
+          #{IPSocket.getaddress(Socket.gethostname)} \
```

Comment:
`Socket.gethostname` returns the FQDN of the, that is in `/etc/hostname`, and the `IPSocket.getaddress` will return the IP defined for this host in `/etc/hosts`. And AFAIK in this file you cannot specify multiple IP address for the same host, so there is only one.

The goal here is to find the IP of the admin machine, and this is where this code is running